src/main/java/hexgui/gui/HexGui.java

```diff
@@ -0,0 +1,2393 @@
+        cb =
+            new Runnable() {
+              public void run() {
+                cbEditParameters();
+              }
+            };
+        break;
+      case VC:
+        cb =
+            new Runnable() {
+              public void run() {
+                cbVCs();
+              }
+            };
+        break;
+      case STRING:
+        cb =
+            new Runnable() {
+              public void run() {
+                cbString();
+              }
+            };
+        break;
+      case VAR:
+        cb =
+            new Runnable() {
+              public void run() {
+                cbVar();
+              }
+            };
+        break;
+    }
+    // if (c.equals("dfpn-get-bounds"))
+    //     cb = new Runnable() { public void run() { cbDfpnDisplayBounds();} };
+    // else if (c.equals("book-scores"))
+    //     cb = new Runnable() { public void run() { cbDisplayBookScores(); } };
+    // else if (c.equals("eval-resist"))
+    //     cb = new Runnable() { public void run() { cbEvalResist(); } };
+    Runnable callback = null;
+    if (cb != null) callback = new GuiRunnable(cb);
+    sendCommand(cmd + "\n", callback);
+  }
+
+  /**
+   * HtpShell Callback. By the name of the command it choose the proper callback function. Arguments
+   * are passed as given.
+   */
+  public void commandEntered(String cmd) {
+    sendCommand(cmd, null);
+  }
+
+  // ----------------------------------------------------------------------
+
+  private boolean commandNeedsToLockGUI(String cmd) {
+    if ((cmd.length() > 7 && cmd.substring(0, 7).equals("genmove"))
+        || (cmd.length() > 15 && cmd.substring(0, 15).equals("dfs-solve-state"))
+        || (cmd.length() > 16 && cmd.substring(0, 16).equals("dfpn-solve-state"))
+        || (cmd.length() > 23 && cmd.substring(0, 23).equals("dfs-solver-find-winning"))
+        || (cmd.length() > 24 && cmd.substring(0, 24).equals("dfpn-solver-find-winning")))
+      return true;
+    return false;
+  }
+
+  private void lockGUI() {
+    m_locked = true;
+    m_toolbar.lockToolbar();
+  }
+
+  private void unlockGUI() {
+    m_toolbar.unlockToolbar(m_current, this);
+    m_locked = false;
+  }
+
+  /** A (command, callback) pair. */
+  private class HtpCommand {
+    public HtpCommand() {}
+
+    public HtpCommand(String cmd, Runnable callback) {
+      this.str = cmd;
+      this.callback = callback;
+    }
+
+    public String str;
+    public Runnable callback;
+  }
+
+  /** Waits for commands to be added to the queue, then processes each in turn. */
+  private class CommandHandler implements Runnable {
+
+    public CommandHandler(Component parent, ArrayBlockingQueue<HtpCommand> queue) {
+      m_parent = parent;
+      m_queue = queue;
+    }
+
+    public void run() {
+      while (true) {
+        HtpCommand cmd = null;
+        try {
+          // block until queue contains an element
+          cmd = m_queue.take();
+        } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)